### DISCUSSION TITLE
Resize block device on flatcar instance (without reboot)

### DISCUSSION THREAD

**Discussion by till** [flatcar/Flatcar#1423] [Q&A] [answered]
Hi,

we use OpenStack with Flatcar, I can resize a volume via OpenStack just fine, but then need to figure out how to make the OS recognize the new size. I currently do a sequence of un-mounting the block storage (using the systemd unit), then running `sudo e2fsck -f /dev/abc; sudo resize2fs /dev/abc` and mounting it again.

Is there a (better or more modern) way to do it all online? Without unmounting, etc..

Thanks!

**Comment by pothos** [comment]
Normally ext4 and xfs can do online growing, and btrfs can do online growing and shrinking, see here for a full list:
https://github.com/storaged-project/libblockdev/blob/313980ff123e0f6abfaac17fba0d510e53b4eda7/src/plugins/fs/generic.c#L60

**Reply by till** [reply]
Are you suggesting to skip the `e2fsck`? I think it's frequently suggested by `resize2fs` to run that. Not sure where I got it otherwise.

**Reply by pothos** [reply, marked as answer]
Yes, and no unmounting as well. Details are in the man page:
```
If the file system is mounted, it can be used to expand the size of the mounted file system, assuming
       the kernel and the file system supports on-line resizing.  (Modern Linux 2.6 kernels will support on-line resize for file  sys‐
       tems  mounted  using  ext3  and  ext4; ext3 file systems will require the use of file systems with the resize_inode feature en‐
       abled.)
```

**Reply by till** [reply]
Ok, resizing worked online.

But the volume extension fails with gophercloud. Will investigate that. 😆

Thank you for helping!
